File: common/src/test/java/guepardoapps/lucahome/common/services/UserService/IsValidPasswordUnitTest.java
package guepardoapps.lucahome.common.services.UserService;

import org.junit.experimental.theories.DataPoints;
import org.junit.experimental.theories.Theories;
import org.junit.experimental.theories.Theory;
import org.junit.runner.RunWith;

import guepardoapps.lucahome.common.services.user.UserService;

import static org.junit.Assert.assertEquals;

@RunWith(Theories.class)
public class IsValidPasswordUnitTest {

    public static class PasswordValidation {
        String password;
        boolean isValid;

        PasswordValidation(String password, boolean isValid) {
            this.password = password;
            this.isValid = isValid;
        }
    }

    @DataPoints
    public static PasswordValidation[] passwords = {
            new PasswordValidation("Pass1", false),
            new PasswordValidation("Pass12", true),
            new PasswordValidation("Pass123", true),
            new PasswordValidation("Password123456789Abc1234567890Ab", true),
            new PasswordValidation("Password123456789Abc1234567890Abc", false),
            new PasswordValidation("Password123456789Abc1234567890Abc1", false),
            new PasswordValidation("Pass123.", false)
    };

    @Theory
    public void isValidUserName_isCorrect(PasswordValidation passwordValidation) throws Exception {
        boolean actualIsValid = UserService.Companion.getInstance().isValidPassword(passwordValidation.password);
        assertEquals(passwordValidation.isValid, actualIsValid);
    }
}

File: common/src/test/java/guepardoapps/lucahome/common/services/user/IsValidPasswordUnitTest.kt
package guepardoapps.lucahome.common.services.user

import org.jetbrains.spek.api.Spek
import org.jetbrains.spek.api.dsl.describe
import org.jetbrains.spek.api.dsl.it
import org.junit.experimental.theories.DataPoints
import org.junit.experimental.theories.Theories
import org.junit.experimental.theories.Theory
import org.junit.runner.RunWith

import org.junit.Assert.assertEquals
import org.junit.platform.runner.JUnitPlatform

// http://spekframework.github.io/spek/docs/latest/#_ignoring_tests

@RunWith(JUnitPlatform::class)
class UserServiceUnitTest : Spek({

    describe("Unit tests for UserServiceUnitTest") {

        beforeEachTest { }

        afterEachTest { }

        it("") {
            // Arrange

            // Act

            // Assert
        }
    }
})
@RunWith(Theories::class)
class IsValidPasswordUnitTest {

    data class PasswordValidation(var password: String, var isValid: Boolean)

    companion object {
        @DataPoints
        var passwords = arrayOf(
                PasswordValidation("Pass1", false),
                PasswordValidation("Pass12", true),
                PasswordValidation("Pass123", true),
                PasswordValidation("Password123456789Abc1234567890Ab", true),
                PasswordValidation("Password123456789Abc1234567890Abc", false),
                PasswordValidation("Password123456789Abc1234567890Abc1", false),
                PasswordValidation("Pass123.", false)
        )
    }

    @Theory
    @Throws(Exception::class)
    fun isValidUserName_isCorrect(passwordValidation: PasswordValidation) {
        val actualIsValid = UserService.instance.isValidPassword(passwordValidation.password)
        assertEquals(passwordValidation.isValid, actualIsValid)
    }
}